app/models/user.rb

```diff
@@ -9,4 +9,14 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :sent_requests, foreign_key: 'sender_id', class_name: 'Friendship'
+  has_many :received_requests, foreign_key: 'receiver_id', class_name: 'Friendship'
```

Comment:
You missed a ver important association here!! You don't have the `has_many :friendships` and the opposite association that could be called `has_many :inverse_friendships`

This prevents the app from working properly since the user can now send an invitation to another (with the button you created) but the logic behind that is not implemented correctly, then the friendship is not created.

See requirements [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friend.png) and [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friend.png)